src/monodroid/jni/debug.cc

```diff
@@ -409,55 +400,68 @@ Debug::handle_server_connection (void)
  *
  *   Process a command received from XS through a socket connection.
  * This is called on a separate thread.
- * Return TRUE, if a new connection need to be opened.
+ * Return true, if a new connection need to be opened.
  */
-int
+bool
 Debug::process_cmd (int fd, char *cmd)
 {
-	if (!strcmp (cmd, "connect output")) {
+	if (strcmp (cmd, "connect output") == 0) {
 		dup2 (fd, 1);
 		dup2 (fd, 2);
-		return TRUE;
-	} else if (!strcmp (cmd, "connect stdout")) {
+		return true;
+	}
+
+	if (strcmp (cmd, "connect stdout") == 0) {
 		dup2 (fd, 1);
-		return TRUE;
-	} else if (!strcmp (cmd, "connect stderr")) {
+		return true;
+	}
+
+	if (strcmp (cmd, "connect stderr") == 0) {
 		dup2 (fd, 2);
-		return TRUE;
-	} else if (!strcmp (cmd, "discard")) {
-		return TRUE;
-	} else if (!strcmp (cmd, "ping")) {
+		return true;
+	}
+
+	if (strcmp (cmd, "discard") == 0) {
+		return true;
+	}
+
+	if (strcmp (cmd, "ping") == 0) {
 		if (!utils.send_uninterrupted (fd, const_cast<void*> (reinterpret_cast<const void*> ("pong")), 5))
 			log_error (LOG_DEFAULT, "Got keepalive request from XS, but could not send response back (%s)\n", strerror (errno));
-	} else if (!strcmp (cmd, "exit process")) {
-		log_info (LOG_DEFAULT, "XS requested an exit, will exit immediately.\n");
+		return false;
+	}
+
+	if (strcmp (cmd, "exit process") == 0) {
+		log_info (LOG_DEFAULT, "Debugger requested an exit, will exit immediately.\n");
 		fflush (stdout);
 		fflush (stderr);
 		exit (0);
-	} else if (!strncmp (cmd, "start debugger: ", 16)) {
+	}
+
+	bool use_fd = false;
+	if (strncmp (cmd, "start debugger: ", 16) == 0) {
```

Comment:
Question: *stylistically* would it be better if we *removed* the hardcoded length and instead went with string constants & `sizeof()`?

```cpp
static constexpr char COMMAND_START_DEBUGGER[] = "start debugger: ";
if (strncmp (cmd, COMMAND_START_DEBUGGER, sizeof (COMMAND_START_DEBUGGER)-1) == 0) {
	...
}
```

Reply:
Yeah, it would be better (even better with `constexpr size_t COMMAND_START_DEBUGGER_LEN = sizeof(COMMAND_START_DEBUGGER)-1` but it would blow out the code size...)

Reply:
but if you don't mind the visual bloat, I'm ok with it :)

Reply:
I think the "visual bloat" would be a good idea, considering that all of these lines are changing *anyway*..,.